packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java

```diff
@@ -623,10 +623,13 @@ public void maybeSetText(ReactTextUpdate reactTextUpdate) {
     if (reactTextUpdate.getText().length() == 0) {
       setText(null);
     } else {
-      // When we update text, we trigger onChangeText code that will
-      // try to update state if the wrapper is available. Temporarily disable
-      // to prevent an infinite loop.
+      boolean shouldKeepCompositeSpan = length() == spannableStringBuilder.length();
```

Comment:
This means that we only retain composing spans if JS wants to set a new string, and the string is the same length as the one currently held in native.

Is this a heuristic to try to preserve location of the spans? I could imagine some real-world cases where this would bail (e.g. we add a dash in a phone number), where in the problem described, we may still want to preserve the composing span. But I understand this gets a bit hairy.

Reply:
Yes, it's a bit heuristic way of preserving the span. 
I agree this could be improved to handle more cases, I just tried to keep focus on this specific issue since I am not that familiar with the codebase.

Note that, even if this solution loses the span, the heuristic fails, the code will be working just as it is working currently without the changes introduced